packages/web3torrent/src/pages/file/File.tsx

```diff
@@ -33,6 +46,19 @@ const File: React.FC<Props> = props => {
   const [loading, setLoading] = useState(false);
   const [buttonLabel, setButtonLabel] = useState('Start Download');
   const [errorLabel, setErrorLabel] = useState('');
+  const [warning, setWarning] = useState('');
+
+  useEffect(() => {
+    const testResult = async () => {
+      const torrentCheckResult = await checkTorrent(web3Torrent, infoHash);
+      setWarning(torrentCheckResult);
+    };
+
+    if (infoHash) {
+      testResult();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Should this effect get executed every time `web3Torrent` or `infoHash` changes? Why wouldn't it?

Reply:
this should be executed once, and just once, at the moment that the client has the infoHash.

On the seeder side, it allows the user to know if the torrent was announced to the tracker.
On the leecher side, it allows the user to know if it could communicate with the tracker, and if the tracker came up with any peers that could seed the file.

Reply:
Got it. But what about when the `web3Torrent` client changes? When the `web3Torrent` client is updated (which doesn't happen in the code currently, but could in the future), shouldn't the `testResult` get retriggered?

What about when the `infoHash` is update? Isn't the `testResult` invalid in that case as well?

Maybe a bigger point is the following. The linter error is saying: 
"Hey, `web3Torrent` and `infoHash` are variables that can change when the component is re-rendered. Are you sure you don't want to rerun the effect when these things change?"

It strikes me that we are ignoring this linter error because we state: "We know better. `web3Torrent` and `infoHash never change". But that's an assumption made in the codebase at this instant. What is the downside of including `web3Torrent` and `hashInfo` in the effect second argument array?

Reply:
The test doesn't check for anything regarding the `webtorrent` client at the moment. Nor the web3torrent client.
The torrent check could be a utility function, as it doesn't depend in any way on the webtorrent client (and now that I think of it, I'm wondering why I thought it was a good idea to put it there... I blame sleepiness, but I can move it to another file so that the webtorrent client doesn't get more cluttered).

About infoHash updates. I was being careful, in order to avoid running 2 times this function at the same time, thus creating 2 trackerClients. I agree that adding infoHash doesn't have a negative consequence. But to be fair, I don't expect the infoHash to ever change. Once the value is set, I expect to be constant through the whole life of this component. Remember that the context of this page is a single torrent, and the infoHash is, in a way, a torrent's id

Reply:
Ok that makes a lot more sense, thank you for explanation 🙏! I think removing the `web3Torrent` dependency clear up my mental model.

Reply:
cool. Thank you too!. It helped me too, to realize that I should move the function to another file so that the web3torrent client doesn't get cluttered with stuff that it doesn't make sense to be there.

I think my reasoning at the time was that _" because the global checker was there, this should be there too"_

Reply:
https://github.com/statechannels/monorepo/pull/1449/commits/94c9a9125ce5465046ebb16f4a2eb594017087d1